draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -204,58 +204,31 @@ Simple Password Exponential Key Exchange {{Jablon96}}, Password Authenticated
-to this process as "serialization", since it is typically used for compactly storing and
-transporting points, or for producing canonicalized outputs. Since a deserialization
-algorithm can often be used as a type of encoding algorithm, we also briefly
-document properties of these functions.
-
-It is often the case that the output of the encoding function should be
-distributed uniformly at random on the elliptic curve. That is, there is no
-discernible relation existing between outputs that can be computed based on
-the inputs. In practice, this requirement stems from needing a random oracle
-which outputs elliptic curve points:  one way to construct this is by first
-taking a regular random oracle, operating entirely on bitstrings, and applying
-a suitable encoding function to the output.
-
-This motivates the term "hashing to the curve", since cryptographic hash
-functions are typically modeled as random oracles. However, this still leaves open
-the question of what constitutes a suitable encoding method, which is a primary
-concern of this document.
-
-A random oracle onto an elliptic curve can also be instantiated using
-direct constructions, however these tend to rely on many group operations
-and are less efficient than hash and encode methods.
-
-This document specifies several algorithms for mapping arbitrary inputs to elliptic
-curve points with varying properties: Icart, SWU, Simplified SWU, and Elligator2.
-For each algorithm, we identify suitable use cases, with a particular emphasis
-on hashing to the curve, to produce a random oracle.
-Each algorithm conforms to a common  interface, i.e., it maps an element from 
-a base field F to a curve E. For each variant, we 
-describe the requirements for F and E to make it work. Sample code for each variant is 
-presented in the appendix.  Unless otherwise stated, all elliptic curve points are assumed to 
-be represented as affine coordinates, i.e., (x, y) points on a curve. 
+Unfortunately for implementors, the precise mapping which is suitable for a
+given scheme is not necessarily included in the description of the protocol.
+Compounding this problem is the need to pick a suitable curve for the specific
+protocol.
+
+This document aims to address this lapse by providing a thorough set of
+recommendations across a range of implementations, and curve types. We provide
+implementation and perfomance details for each mechanism, along with references
+to the security rationale behind each recommendation and guidance for
+applications not yet covered.
+
+Each algorithm conforms to a common interface, i.e., it maps an element from a
+base field F to a curve E. For each variant, we describe the requirements for F
+and E to make it work. Sample code for each variant is presented in the
+appendix.  Unless otherwise stated, all elliptic curve points are assumed to be
+represented as affine coordinates, i.e., (x, y) points on a curve.
```

Comment:
I was thinking of adding tables to the recommendations along these line:

The following table lists algorithms recommended by use-case:

| Application       | Requirement   | Additional Details
|-------------------|---------------|---------|
| SPEKE {{Jablon96}}| Naive         | y = H(x)*G |
| PAKE  {{BMP00}}   | Random Oracle |   -    | 
| BLS {{BLS01}}     | RO            |    -   |
| IBE {{BF01}}      | RO | Supersingular, pairing-friendly curve |

To find the suitable algorithm, lookup the requirement from above, with 
the chosen curve in the below:


| Curve  | Inj. Encoding | RO |
|--------|---------------|------|
| P-256 | SWU {{simple-swu}} | FFSTV(SWU)
| P-384 | Icart {{icart}} | FFSTV(Icart)
| Curve25519 | Elligator2 {{elligator2}} | ...
| Curve448 | Elligator2 {{elligator2}} | ...

----

Reply:
Which will hopefully be more obviously useful when expanding the list of applications to tease out the subtle requirements (functional and security). For example, I think we can show that the x-only Montgomery mechanism (which gives you an injective encoding) is useful for many applications where you need a PRF-like thing (potentially relevant to @grittygrease in fact).

Reply:
As for the structure of encoding & hashing vs all algorithms. I don't think there will be a huge difference, since AFAICT all mechanism to do RO properly are generic methods using some existing injective encoding. So I would present all encoding mechanism and mention their specific properties, and then list the methods to achieve random oracles by combining them.

Reply:
+1 to all of the above. :-)